src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -214,24 +214,26 @@ private static CastResult TryGet(nuint source, nuint target)
                 if (interfaceCount != 0)
                 {
                     MethodTable** interfaceMap = mt->InterfaceMap;
-                    for (nuint i = 0; ; i += 4)
+                    nuint i = 0;
+                    for (; i < interfaceCount - 4; i += 4)
                     {
-                        if (interfaceMap[i + 0] == toTypeHnd)
+                        MethodTable** map = interfaceMap + 4;
+                        if (interfaceMap[0] == toTypeHnd ||
+                            interfaceMap[1] == toTypeHnd ||
+                            interfaceMap[2] == toTypeHnd ||
+                            interfaceMap[3] == toTypeHnd)
                             goto done;
-                        if (--interfaceCount == 0)
-                            break;
-                        if (interfaceMap[i + 1] == toTypeHnd)
-                            goto done;
-                        if (--interfaceCount == 0)
-                            break;
-                        if (interfaceMap[i + 2] == toTypeHnd)
-                            goto done;
-                        if (--interfaceCount == 0)
-                            break;
-                        if (interfaceMap[i + 3] == toTypeHnd)
+
+                        interfaceMap = map;
+                    }
+
+                    for (; i < interfaceCount; i++)
+                    {
+                        MethodTable** map = interfaceMap + 1;
```

Comment:
looks like this skips the very first `interfaceMap[0]` item doesn't it?

Reply:
Shouldn't? The `interfaceMap + 1` assignment to `map` is used for a zero-latency MOV after the test of `interfaceMap[0]`
```csharp
MethodTable** map = interfaceMap + 1;
if (interfaceMap[0] == toTypeHnd)
    goto done;

interfaceMap = map;
```

Reply:
unsigned subtraction is catching me out though 😅